pkg/api/types.go

```diff
@@ -607,6 +607,10 @@ type SecretVolumeSource struct {
 	// the volume setup will error. Paths must be relative and may not contain
 	// the '..' path or start with '..'.
 	Items []KeyToPath `json:"items,omitempty"`
+	// Optional: mode bits to use on created files by default. The final
+	// mode bits will be the provided value AND 0777. Defaults to 0644.
```

Comment:
On Wed, Aug 03, 2016 at 08:52:12PM -0700, Tim Hockin wrote:
> > @@ -607,6 +607,10 @@ type SecretVolumeSource struct {
> >  	// the volume setup will error. Paths must be relative and may not contain
> >  	// the '..' path or start with '..'.
> >  	Items []KeyToPath `json:"items,omitempty"`
> > +	// Optional: mode bits to use on created files by default. The final
> > +	// mode bits will be the provided value AND 0777. Defaults to 0644.
> 
> If we are validating that this is < 0777, then the comments about AND  can be removed.  Just spec that it must be between 0 and 0777

I had just realized about this too, while looking at the patch as a whole. I'll
clean things up (things like this, probably something else is there too), and add
tests during the weekend :-)